addons/product/pricelist.py

```diff
@@ -252,86 +178,72 @@ def _price_rule_get_multi(self, cr, uid, pricelist, products_by_qty_by_partner,
-                            if qty_uom_id != seller_uom:
-                                qty_in_seller = product_uom_obj._compute_qty(cr, uid, qty_uom_id, qty, to_uom_id=seller_uom)
-                            for line in seller.pricelist_ids:
-                                if line.min_quantity <= qty_in_seller:
-                                    price = line.price
-
-                if price is not False:
-                    price_limit = price
-                    price = price * (1.0+(rule.price_discount or 0.0))
-                    if rule.price_round:
-                        price = tools.float_round(price, precision_rounding=rule.price_round)
-
-                    convert_to_price_uom = (lambda price: product_uom_obj._compute_price(
-                                                cr, uid, product.uom_id.id,
-                                                price, price_uom_id))
-                    if rule.price_surcharge:
-                        price_surcharge = convert_to_price_uom(rule.price_surcharge)
-                        price += price_surcharge
-
-                    if rule.price_min_margin:
-                        price_min_margin = convert_to_price_uom(rule.price_min_margin)
-                        price = max(price, price_limit + price_min_margin)
-
-                    if rule.price_max_margin:
-                        price_max_margin = convert_to_price_uom(rule.price_max_margin)
-                        price = min(price, price_limit + price_max_margin)
-
-                    rule_id = rule.id
-                break
-
+                        price = currency_obj.compute(
+                                cr, uid,
+                                pricelist.company_id.currency_id.id, pricelist.currency_id.id,
+                                product_obj._price_get(cr, uid, [product], rule.base, context=context)[product.id],
+                                round=False, context=context)
+                    if price is not False:
+                        if rule.compute_price == 'fixed':
+                            price = rule.fixed_price
+                        elif rule.compute_price == 'percentage':
+                            price = (price - (price * (rule.percent_price/100))) or 0.0
+                        else:
+                            price_limit = price
+                            price = price * (1.0+(rule.price_discount or 0.0))
+                            if rule.price_round:
+                                price = tools.float_round(price, precision_rounding=rule.price_round)
+
+                            convert_to_price_uom = (lambda price: product_uom_obj._compute_price(
+                                                        cr, uid, product.uom_id.id,
+                                                        price, price_uom_id))
+                            if rule.price_surcharge:
+                                price_surcharge = convert_to_price_uom(rule.price_surcharge)
```

Comment:
why need to convert surcharge, price_min_margin, price_max_margin amount into final uom ?

I think, only need to convert final price amount in this line: https://github.com/odoo-dev/odoo/blob/master-simplify-pricelists-rmu/addons/product/pricelist.py#L246